### PR TITLE
fix: host preflight checks results output colored

### PR DESCRIPTION
**Pull request by camilamacedo86**
#### What this PR does / why we need it:

Currently, because of the usage of the tee to generate the logs we are unable to check the host preflights results with colours which makes harder to identify what is an error, warning and etc. 

#### Which issue(s) this PR fixes:

Fixes #

#### Special notes for your reviewer:

See:

Screenshot 2023-03-06 at 12 30 12

## Steps to reproduce
Just run the install command and check the host preflights checks result. 

#### Does this PR introduce a user-facing change?

```release-note
fix: host preflight checks result by outputting them with colours in order to improve the user experience
```

#### Does this PR require documentation?
NONE

### CODE REVIEW

File: pkg/cli/host_preflight.go [hunk 1]
```diff
@@ -137,7 +137,7 @@ func newHostPreflightCmd(cli CLI) *cobra.Command {
 				return errors.Wrap(err, "run host preflight")
 			}
 
-			printPreflightResults(cmd.OutOrStdout(), results)
```
Comment (laverya) [inline pkg/cli/host_preflight.go:140]: why the change to no longer use outorstdout?

I feel like this is the change that's actually making a difference

Comment (camilamacedo86) [reply pkg/cli/host_preflight.go]: Yep, I agree I reverted that.

File: pkg/cli/host_preflight_test.go [hunk 2]
```diff
@@ -45,7 +45,7 @@ func TestNewHostPreflightCmd(t *testing.T) {
 					IsPass:  true,
 				},
 			},
-			stdout: green("[PASS]") + " Number of CPUs: At least 4 CPU cores are required\n",
+			stdout: OutputPassGreen() + " Number of CPUs: At least 4 CPU cores are required\n",
```
Comment (laverya) [inline pkg/cli/host_preflight_test.go:48]: I would GREATLY prefer for this not to call `OutputPassGreen` etc

Comment (laverya) [reply pkg/cli/host_preflight_test.go:48]: A() = A() isn't a good unit test IMO

Comment (camilamacedo86) [reply pkg/cli/host_preflight_test.go:48]: That is how that was before. I think has no point to change the tests now.

**Review by laverya — APPROVED** [review]

File: go.mod [hunk 3]
```diff
@@ -319,6 +319,7 @@ replace (
 	k8s.io/controller-manager => k8s.io/controller-manager v0.26.1
 	k8s.io/cri-api => k8s.io/cri-api v0.26.1
 	k8s.io/csi-translation-lib => k8s.io/csi-translation-lib v0.26.1
+	k8s.io/dynamic-resource-allocation => k8s.io/dynamic-resource-allocation v0.26.1
```
Comment (camilamacedo86) [inline go.mod:322]:
```suggestion
```
